### PR TITLE
Adding README for Microsoft.Extensions.EnumStrings

### PR DESCRIPTION
**Pull request by joperezr**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4621)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.EnumStrings/README.md [hunk 1]
```diff
@@ -1 +1,80 @@
-README
+# Microsoft.Extensions.EnumStrings
+
+It's often necessary to convert a C# enum value to a string. The standard ToString() is normally used for this, but that can not be the best in terms of allocations or performance. This package introduces a source generator that is triggered by an attribute which produces an extension method for each enum type annotated with the attribute which can efficiently convert an enum value to a string.
```
Comment (sebastienros) [inline src/Libraries/Microsoft.Extensions.EnumStrings/README.md:3]:
```suggestion
It's often necessary to convert a C# enum value to a string. The standard `ToString()` method is normally used for this, but that can not be the best in terms of allocations or performance. This package introduces a source generator that is triggered by an attribute which produces an extension method for each `enum` type annotated with the attribute which can efficiently convert an enumeration value to a string.
```

Pick whatever suggestion

File: src/Libraries/Microsoft.Extensions.EnumStrings/README.md [hunk 2]
```diff
@@ -1 +1,80 @@
...
+```
+
+## Usage Example
+
+### Adding the attribute to an enum
+
+To use this feature, you can annotate an enum value with the `EnumStringAttribute` attribute. For example:
+
+```csharp
+using Microsoft.Extensions.EnumStrings;
```
Comment (joperezr) [inline src/Libraries/Microsoft.Extensions.EnumStrings/README.md:28]: @geeknoid should we consider moving the EnumStrings attribute to the usual `System.Runtime.CompilerServices` namespace where we put the rest of the attributes in .NET in favor of a future move of this down to dotnet/runtime?

Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.EnumStrings/README.md:28]: cc: @terrajobst @stephentoub

Comment (geeknoid) [reply src/Libraries/Microsoft.Extensions.EnumStrings/README.md:28]: We certainly can.

**Review by geeknoid — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.EnumStrings/README.md [hunk 3]
```diff
@@ -1 +1,80 @@
...
+    var color = Color.Red;
+
+    var s1 = color.ToString(); // <- This line will cause an allocation and won't cache the result.
+    var s2 = color.ToInvariantString(); // <- This call is more performant and won't cause a new allocation.
+}
+```
+
+Both calls will return the same string, but the second one will be more efficient.
+
+### Using the generated with an enum from a different assembly
```
Comment (Tratcher) [inline src/Libraries/Microsoft.Extensions.EnumStrings/README.md:61]:
```suggestion
### Using the generator with an enum from a different assembly
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**